internal/aws/awsutil/conn.go

```diff
@@ -203,29 +203,51 @@ func GetAWSConfigSession(logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSetting
+			Providers: customCredentialProvider(cfg, config, s),
 		})
 	}
 	return config, s, nil
 }
 
-func customCredentialProvider(cfg *AWSSessionSettings, config *aws.Config) []credentials.Provider {
-	defaultCredProviders := defaults.CredProviders(config, defaults.Handlers())
+// order
+// 1. shared profile
+// 2. env var
+// 3. shared creds file
+// 4. ec2 creds
+// 5. override creds providers
+// explicit keys do not make sense in this context since contrib does not take in explicit keys
+// and cwa does not provide explicit keys to config but come from these methods of getting creds
+// @TODO make this more upstream friendly working in the future
+func customCredentialProvider(cfg *AWSSessionSettings, config *aws.Config, sess *session.Session) []credentials.Provider {
 	overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
 	credProviders := make([]credentials.Provider, 0)
-	// if is for differently configured shared creds file location
-	// else if is for diff profile but no change in creds file ex run in containers
-	if cfg.SharedCredentialsFile != nil && len(cfg.SharedCredentialsFile) > 0 {
+	// this is to keep ordering where if there is a profile we want to try those creds first
+	if cfg.Profile != "" && cfg.SharedCredentialsFile != nil && len(cfg.SharedCredentialsFile) > 0 {
 		for _, file := range cfg.SharedCredentialsFile {
-			credProviders = append(credProviders, &credentials.SharedCredentialsProvider{Filename: file, Profile: cfg.Profile})
+			sharedCredentialsProvider := credentials.SharedCredentialsProvider{Filename: file, Profile: cfg.Profile}
+			credProviders = append(credProviders, &sharedCredentialsProvider)
+			credProviders = append(credProviders, &RefreshableSharedCredentialsProvider{sharedCredentialsProvider: &sharedCredentialsProvider})
 		}
 	} else if cfg.Profile != "" {
-		credProviders = append(credProviders, &credentials.SharedCredentialsProvider{Filename: "", Profile: cfg.Profile})
+		sharedCredentialsProvider := credentials.SharedCredentialsProvider{Filename: "", Profile: cfg.Profile}
+		credProviders = append(credProviders, &sharedCredentialsProvider)
+		credProviders = append(credProviders, &RefreshableSharedCredentialsProvider{sharedCredentialsProvider: &sharedCredentialsProvider})
 	}
-	credProviders = append(credProviders, defaultCredProviders...)
+	credProviders = append(credProviders, &credentials.EnvProvider{})
+	if cfg.Profile == "" && cfg.SharedCredentialsFile != nil && len(cfg.SharedCredentialsFile) > 0 {
+		for _, file := range cfg.SharedCredentialsFile {
+			sharedCredentialsProvider := credentials.SharedCredentialsProvider{Filename: file, Profile: cfg.Profile}
+			credProviders = append(credProviders, &sharedCredentialsProvider)
+			credProviders = append(credProviders, &RefreshableSharedCredentialsProvider{sharedCredentialsProvider: &sharedCredentialsProvider})
+		}
+	}
+	sharedCredentialsProvider := credentials.SharedCredentialsProvider{Filename: "", Profile: ""}
```

Comment:
What is this for? Is this for default profile?

Reply:
Looks like the default profile we use for onprem is diff, `AmazonCloudWatchAgent`: https://github.com/aws/amazon-cloudwatch-agent/blob/46146905924b34093de4dfda572372f7621c560a/translator/util/sdkutil.go#L170

Reply:
Yes this is the default